### PR TITLE
Remove gen_get_params & gen_gettable_params from keygen operation

### PR DESCRIPTION
**Pull request by slontis**
EVP_PKEY_CTX_gettable_params() was missing code for the keygen operation.
After adding it it was noticed that it is probably not required for this type, so instead
the gen_get_params and gen_gettable_params have been remnoved from the provider interface.
gen_get_params was only implemented for ec to get the curve name. This seems redundant
since normally you would set parameters into the keygen_init() and then generate a key.
Normally you would expect to extract data from the key - not the object that we just set up
to do the keygen.

Added a simple settable and gettable test into a test that does keygen.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by levitte** [comment]
This is incomplete. You need to deal with the SM2 hack in crypto/evp/pmeth_gn.c as well.

**Comment by slontis** [comment]
> This is incomplete. You need to deal with the SM2 hack in crypto/evp/pmeth_gn.c as well.

Ahh ok. Updated the hack :)

File: crypto/evp/pmeth_lib.c [hunk 1]
```diff
@@ -656,8 +648,7 @@ const OSSL_PARAM *EVP_PKEY_CTX_settable_params(EVP_PKEY_CTX *ctx)
             && ctx->op.ciph.cipher->settable_ctx_params != NULL)
         return ctx->op.ciph.cipher->settable_ctx_params();
     if (EVP_PKEY_CTX_IS_GEN_OP(ctx)
-            && ctx->keymgmt != NULL
-            && ctx->keymgmt->gen_settable_params != NULL)
+            && ctx->keymgmt != NULL)
```
Comment (levitte) [inline crypto/evp/pmeth_lib.c:651]: Er, why?

Comment (slontis) [reply crypto/evp/pmeth_lib.c:651]: I knew you would ask that - the util function that it calls does this check. It is different to some of the others that just call the function directly if the value is not NULL.

Comment (levitte) [reply crypto/evp/pmeth_lib.c:651]: Ah!

**Comment by slontis** [comment]
Thanks. Merged to master.

**Pull request closed without merging**